### PR TITLE
Update iOS and Android dependencies to Firebase 8.0.0, except for AdMob.

### PR DESCRIPTION
**Pull request by jonsimantov**
Requires custom AdMob version, currently set to 7.69.0-cppsdk.

### CODE REVIEW

**Comment by jonsimantov** [comment]
Packaging run: https://github.com/firebase/firebase-cpp-sdk/actions/runs/908067621

**Comment by jonsimantov** [comment]
@ehsannas Any idea why some Firestore tests are failing? Are these flakes?

**Comment by ehsannas** [comment]
@jonsimantov I'm not sure why these are failing, and AFAIK we're not seeing such flakes on `main`.

@var-const @dconeybe @wu-hui Have you seen any of these tests fail recently?

**Comment by dconeybe** [comment]
> @var-const @dconeybe @wu-hui Have you seen any of these tests fail recently?

Those failures do not look familiar to me. Every failure, however, seems to be a timeout waiting for a `Future` to complete. This smells like a flake and I'd recommend re-running before spending too much time investigating.

**Comment by var-const** [comment]
> @var-const @dconeybe @wu-hui Have you seen any of these tests fail recently?

I also haven't seen these fail recently.

File: release_build_files/Android/firebase_dependencies.gradle [hunk 1]
```diff
@@ -16,30 +16,30 @@ import org.gradle.util.ConfigureUtil;
 
 // A map of library to the dependencies that need to be added for it.
 def firebaseDependenciesMap = [
-  'app' : ['com.google.firebase:firebase-analytics:18.0.3'],
+  'app' : ['com.google.firebase:firebase-analytics:19.0.0'],
   'admob' : ['com.google.firebase:firebase-ads:19.8.0',
              'com.google.android.gms:play-services-measurement-sdk-api:18.0.3',
```
Comment (a-maurice) [inline release_build_files/Android/firebase_dependencies.gradle:21]: The other one got changed to analytics:19.0.0, I assume this one needs to do the same

File: admob/integration_test/build.gradle [hunk 2]
```diff
@@ -73,6 +77,10 @@ android {
 apply from: "$gradle.firebase_cpp_sdk_dir/Android/firebase_dependencies.gradle"
 firebaseCpp.dependencies {
   admob
+  messaging
```
Comment (a-maurice) [inline admob/integration_test/build.gradle:80]: I'm cool with leaving these, but we should probably put a comment explaining why

Comment (jonsimantov) [reply admob/integration_test/build.gradle:80]: Oh no I should remove that, it was just for testing.

File: release_build_files/Android/firebase_dependencies.gradle (hunk 1 above)
Comment (jonsimantov) [reply release_build_files/Android/firebase_dependencies.gradle:21]: Yep. Good catch!

**Review by a-maurice — APPROVED** [review]

**Review by a-maurice — APPROVED** [review]

**Pull request merged**